views/AnimatedProfile.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react'
+import { View, Text, Image, StyleSheet, TextInput, Dimensions, Animated, Easing, TouchableOpacity } from "react-native"
+import { connect } from "react-redux"
+import Badges from "../components/Badges"
+
+const { height, width } = Dimensions.get("window")
+
+interface Props {
+  username: string,
+  showProfile: boolean,
+  setUserName: Function,
+  closeProfile: Function
+  currentProfile: User
+}
+
+interface User {
+  first_name: string,
+  profile_photo: string,
+  lvl: string
+}
+
+export class AnimatedProfile extends Component<Props> {
+  componentDidUpdate() {
+    if (this.props.showProfile) {
+      this.yTranslate.setValue(0);
+      Animated.spring(this.yTranslate, {
+        toValue: 1,
+        friction: 6
+      }).start();
+    } else {
+      Animated.timing(this.yTranslate, {
+        toValue: 0,
+        duration: 200,
+        easing: Easing.linear
+      }).start();
+    }
+  }
+
+  yTranslate = new Animated.Value(0)
```

Comment:
yTranslate is declared outside of "render" as it is accessed both within "componentDidUpdate" and "render"